doc/memos/rdm0.md

```diff
@@ -0,0 +1,152 @@
+
+## Terminology
+
+This memo uses the [RFC2119](https://www.ietf.org/rfc/rfc2119.txt) terminology and the following acronyms:
+
+- RIOT-MC: RIOT Maintainers Community
+- RDM: RIOT Developer Memo
+- PR: git Pull-Request to RIOT's master branch on GitHub 
+- ACK: acknowledgement from a RIOT maintainer, concerning a pull-request to the 
+master branch of RIOT on GitHub
+
+# 1. Introduction
+
+In order to facilitate RIOT maintenance in the long term and at large scale, 
+additional documentation complementing code and usual code documentation is 
+needed. For instance, such memos are expected to describe and give explanations 
+about architectural design decisions, code structure etc. Other memos might
+also describe other aspects of RIOT activity, including, but not limited to,
+RIOT community processes, position with respect to some related external
+technical debates, pitfalls and best practices...
+
+
+This memo specifies a format for such memos and the process of discussing their 
+content before publication. Format and processes are somewhat inspired by 
+[IETF](http://ietf.org/) processes and documents such as Request For Comments 
+(RFC).
+
+# 2. RIOT Developers Memo Format
+
+RIOT Developer Memos (RDM) are written in English, using the Markdown (.md) 
+format, with less than 80 characters per line. Each memo has a unique 
+identification number indicated in its header, 
+e.g. this memo is RDM0.
+
+The structure of a memo MUST contain:
+
+1. a preamble as defined below,
+2. the content of the memo structured as needed,
+3. a postamble as defined below.
+
+The preamble of the memo should be structured as such:
+
+1. a header formatted as the header of this memo,
+2. a short abstract,
+3. a status for the memo stating its license and the category of the memo (e.g. 
+that it is a consensus document),
+4. optionally, a terminology section which recalls/disambiguates definition of 
+terms that are key to the memo.
+
+The core content of the memo is structured as deemed necessary. However, during 
+the review process, care must be taken to follow the below principles as much
+as possible:
+
+1. one memo = one problem/aspect,
+2. start by crisply explaining and motivating the one problem/aspect the memo 
+targets,
+3. memos are expected to be of normative and durable nature, hence: keep a memo 
+as concise as possible without impairing clarity, and factor out content that 
+would be likely to be updated too often.
+
+The postamble of the memo should be structured as such:
+
+1. (optional) acknowledgements section thanking contributors other than the 
+authors,
+2. a section with the main references that relate to the content, with a strong
+preference for permanent, self-contained references,
+3. the list of revisions so far and corresponding changelog,
+4. contact information to reach the author(s).
+
+
+# 3. Memo Publishing and Maintenance Process
+
+Each new memo MUST be proposed as a pull-request (PR),
+with temporary number RDM 99999, 
```

Comment:
Why a temporary number? It will make it hard to refer to the memo before it is accepted, especially if it is referred to from other pending memos as well.
Maybe we could start by numbering it 99999 and immediately get a maintainer to assign it a proper number while still in review stage.

Reply:
I would vote to only assign the number at the very end of the review. Otherwise we might have big wholes in our RDM list if an RDM get's rejected after all. Maybe something similar to Internet drafts would be better here: RDM-draft-<author>-<topic>. This way it is at least somewhat referable.

Reply:
@miri64 so you mean "RDM draft" in the meta-data instead of "RDM 99999" ?

Reply:
Or we keep rejected RDM for "memo" reasons. As experienced quite often, it is nice to be able to point to a reference when rejecting changes.

Reply:
> In the mean time RDM5 is published... would look bad.

Do we care about the looks?
Does that matter? Internet RFC's aren't published in numerical order, either.

Reply:
rejected RDM are kept automatically as closed PRs. Is that enough?

Reply:
@kaspar030 actually RFCs *are* published in a way such that there are mostly no holes in the numbers of published RFCs. And: RFC numbers are assigned by the RFC editor, just before publication (not at the time of fiddling with drafts ;)

Reply:
> @miri64 so for this RDM you mean "RDM: draft-baccelli-rdm-format" in the meta-data instead of "RDM 99999" […]?

Yes, e.g, but also the file name being `rdm-draft-baccelli-rdm-format`

> (which it is not using btw ;)?

I RDM0 is a special case here ;-P

Reply:
@miri64 added specification of temporary name à la Internet Draft.

Reply:
Sounds good to me